File: android/src/main/java/au/com/crazybean/mobilex/kurir/dependency/login/LoginModule.kt
package au.com.crazybean.mobilex.kurir.dependency.login

import au.com.crazybean.mobilex.kurir.modules.login.LoginDelegate
import au.com.crazybean.mobilex.kurir.modules.login.LoginViewModel
import au.com.crazybean.mobilex.kurir.modules.login.LoginView
import org.koin.androidx.viewmodel.dsl.viewModel
import org.koin.dsl.module

val loginModule = module {
    // Delegate
    factory { (view: LoginView?) ->
        LoginDelegate(view, get())
    }

    // ViewModel
    viewModel {
        LoginViewModel(get())
    }
}

File: ios/Kurir/modules/auth/login/impl/LoginModule.swift
//
//  LoginModule.swift
//  Kurir
//
//  Created by Loren on 19/8/19.
//  Copyright © 2019 Crazybean Studio. All rights reserved.
//

import Foundation
import Mobilex

class LoginModule: Module {
    override func inject() {
        // ViewModel
        factory(LoginViewModel.self) { r in
            LoginViewModel(userData: r.resolve(UserData.self), repository: r.resolve(UsersRepository.self)!)
        }
        
        // Delegate
        factory(LoginDelegate.self) { (r, view: LoginView) -> LoginDelegate in
            LoginDelegate(view: view, viewModel: r.resolve(LoginViewModel.self)!)
        }
    }
}
